### PR TITLE
New Hormone therapy and period questions

### PR DESCRIPTION
**Pull request by MikeD-ZOE**
# Description

MVP of Hormone therapy questions:

- [X] Create isPeriodCapable on currentPatient
- [X] Populate isPeriodCapable from patient doc
- [X] Populate isPeriodCapable from AboutYou screen
- [X] Created HormoneTherapyQuestion component
- [X] Added to YourHealthScreen and validation
- [X] Wire up to backend fields for the two questions
- [X] Add to ProfileBackDateScreen
- [X] Update navigation logic to go to ProfileBackDateScreen for hormone therapy based questions
- [X] Separate period and hormone therapy questions into two separate field components
- [X] Use feature-flag to hide pregnancy question

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [X] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

Simulator Screen Shot - iPhone 8 Plus - 2020-05-05 at 19 44 47

## Checklist
- [ ] I have updated mockServer

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: src/core/user/UserService.ts [hunk 1]
```diff
@@ -189,6 +192,10 @@ export default class UserService extends ApiClientBase {
     const isReportedByAnother = patient.reported_by_another || false;
     const isSameHousehold = patient.same_household_as_reporter || false;
 
+    // Hormone treatment related questions
+    const hasPeriodAnswer = !isPeriodCapable || !!patient.period_status;
+    const hasHormoneTreatmentAnswer = !isPeriodCapable || patient.hormone_treatments?.length > 0;
```
Comment (JoshElgar) [inline src/core/user/UserService.ts:197]: Hormone treatments could be null, needs a !! Coerce

Comment (MikeD-ZOE) [reply src/core/user/UserService.ts:197]: Good catch!

File: src/features/patient/fields/HormoneTreatmentQuestion.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    }
+    if (this.props.showHormoneTreatmentQuestion === false) {
+      this.setState({ showHormoneTreatment: false });
+    }
+  }
+
+  static initialFormValues = () => {
+    return {
+      havingPeriods: '',
+      hormoneTreatment: [] as string[],
```
Comment (JoshElgar) [inline src/features/patient/fields/HormoneTreatmentQuestion.tsx:128]: Don't think this needs to be a function

File: src/features/assessment/ProfileBackDateScreen.tsx [hunk 3]
```diff
@@ -86,39 +102,60 @@ export default class ProfileBackDateScreen extends Component<BackDateProps, Stat
...
+        infos = {
+          ...infos,
+          takes_any_blood_pressure_medications: formData.takesAnyBloodPressureMedications === 'yes',
+        };
+      }
+
+      if (infos.takes_any_blood_pressure_medications) {
+        infos = {
+          ...infos,
+          takes_blood_pressure_medications: formData.takesBloodPressureMedications === 'yes', // pril
```
[the remark this answers is not in the document]
Comment (JoshElgar) [reply src/features/assessment/ProfileBackDateScreen.tsx:116]: `...(formData.mechanism !== 'other' && { mechanism: formData.mechanism }),` for example

File: src/features/patient/fields/HormoneTreatmentQuestion.tsx (hunk 2 above)
Comment (MikeD-ZOE) [reply src/features/patient/fields/HormoneTreatmentQuestion.tsx:89]: It does. The screen calls this field method to retrieve the initial values. This is part of the field component interface currently.

File: src/features/assessment/ProfileBackDateScreen.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (MikeD-ZOE) [reply src/features/assessment/ProfileBackDateScreen.tsx:137]: IMO that results in less readable code.

Perhaps a better method is to have an array of subDocuments. Then each conditional block of code just pushes a subdocument into this array, then at the end of the function we do a `return Object.assign.apply(null, arrayOfSubDocs)` which will return one merged document.

**Pull request merged**